terraform/decal-vms.tf

```diff
@@ -0,0 +1,99 @@
+variable "do_token" {}
+variable "dnsimple_token" {}
+variable "dnsimple_account" {}
+
+variable "decal_ssh_fingerprint" {
+  type = "list"
+}
+
+variable "students" {
+  type = "list"
+}
+
+variable "berkeley_subnets" {
+  type = "list"
+}
+
+variable "berkeley_subnets6" {
+  type = "list"
+}
+
+provider "digitalocean" {
+  token = "${ var.do_token }"
+}
+
+provider "dnsimple" {
+  token   = "${ var.dnsimple_token }"
+  account = "${ var.dnsimple_account }"
+}
+
+resource "digitalocean_tag" "staff" {
+  name = "staff"
+}
+
+resource "digitalocean_tag" "student" {
+  name = "student"
+}
+
+resource "digitalocean_firewall" "student_firewall" {
+  name = "only-uc-berkeley"
+  tags = ["${ digitalocean_tag.staff.id }", "${ digitalocean_tag.student.id }"]
+
+  inbound_rule = [
+    {
+      protocol         = "tcp"
+      port_range       = "1-65535"
+      source_addresses = "${ concat(var.berkeley_subnets, var.berkeley_subnets6) }"
+    },
+    {
+      protocol         = "udp"
+      port_range       = "1-65535"
+      source_addresses = "${ concat(var.berkeley_subnets, var.berkeley_subnets6) }"
+    },
+    {
+      protocol         = "icmp"
+      port_range       = "0"
+      source_addresses = "${ concat(var.berkeley_subnets, var.berkeley_subnets6) }"
```

Comment:
Maybe allow ICMP from everywhere? Might be misleading I suppose if you see you can ping a machine but then can't connect, but that's what the current OCF setup is.

Reply:
While confusing, I figured it was somewhat safer and simpler to only allow traffic from our subnet. Admittedly the students are going to be confused either way.

Reply:
Yeah, just leaving it like this is fine, I mean ICMP shouldn't matter much when it comes to security unless we're doing some strange stuff, but I agree about this being simpler.